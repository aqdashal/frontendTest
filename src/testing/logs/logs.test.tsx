import React from 'react';
import LogsPage from '../../containers/logs';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('LoggerPage', () => {
    test('should render without crashing', () => {
        render(<LogsPage />);
    })

    test("snapshot", () => {
        const tree = renderer.create(<LogsPage />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
})