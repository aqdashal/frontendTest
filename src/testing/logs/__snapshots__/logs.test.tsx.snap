// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggerPage snapshot 1`] = `
<div
  className="container mt-4"
>
  <nav
    aria-label="breadcrumb"
    className="pb-2"
    style={
      Object {
        "--bs-breadcrumb-divider": ">",
      }
    }
  >
    <ol
      className="breadcrumb"
    >
      <li
        className="breadcrumb-item"
      >
        Home 
        &gt;
      </li>
      <li
        className="breadcrumb-item"
      >
        Administration 
        &gt;
      </li>
      <li
        aria-current="page"
        className="breadcrumb-item text-primary"
      >
        Logger Search
      </li>
    </ol>
  </nav>
  <div
    className="d-flex align-items-end"
  >
    <div
      className="form-group w-100 me-3"
    >
      <label
        htmlFor="empName"
      >
        Employee Name
      </label>
      <input
        className="form-control"
        name="empName"
        onChange={[Function]}
        placeholder="e.g admin user"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group w-100 me-3"
    >
      <label
        htmlFor="actionType"
      >
        Action Type
      </label>
      <select
        aria-label="actionType"
        className="form-control"
        name="actionType"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          --select--
        </option>
      </select>
    </div>
    <div
      className="form-group w-100 me-3"
    >
      <label
        htmlFor="applicationType"
      >
        Application Type
      </label>
      <select
        className="form-control"
        name="applicationType"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          --select--
        </option>
      </select>
    </div>
    <div
      className="form-group w-100 me-3"
    >
      <label
        htmlFor="fromDate"
      >
        From Date
      </label>
      <input
        className="form-control"
        name="fromDate"
        onChange={[Function]}
        type="date"
        value=""
      />
    </div>
    <div
      className="form-group w-100 me-3"
    >
      <label
        htmlFor="toDate"
      >
        To Date
      </label>
      <input
        className="form-control"
        name="toDate"
        onChange={[Function]}
        type="date"
        value=""
      />
    </div>
    <div
      className="form-group w-100 me-3"
    >
      <label
        htmlFor="applicationId"
      >
        Application ID
      </label>
      <input
        className="form-control"
        name="applicationId"
        onChange={[Function]}
        placeholder="eg.173894756234504"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group w-100"
    >
      <label
        htmlFor="searchBtn"
      />
      <button
        className="btn btn-primary d-block w-100"
        name="searchBtn"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="shadow"
  >
    <div
      className="mt-4"
    >
      <table
        className="w-100"
      >
        <thead>
          <tr>
            <th
              className="default border-bottom"
              onClick={[Function]}
            >
              Log ID
              <span>
                <img
                  src="default.png"
                />
              </span>
            </th>
            <th
              className="default border-bottom"
              onClick={[Function]}
            >
              Application Type
              <span>
                <img
                  src="default.png"
                />
              </span>
            </th>
            <th
              className="default border-bottom"
              onClick={[Function]}
            >
              Application ID
              <span>
                <img
                  src="default.png"
                />
              </span>
            </th>
            <th
              className="default border-bottom"
              onClick={[Function]}
            >
              Action
              <span>
                <img
                  src="default.png"
                />
              </span>
            </th>
            <th
              className=" border-bottom"
              onClick={null}
            >
              Action Details
              <span>
                <img />
              </span>
            </th>
            <th
              className="default border-bottom"
              onClick={[Function]}
            >
              Date:Time
              <span>
                <img
                  src="default.png"
                />
              </span>
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
